### PR TITLE
Added additional comments to AppendFormatHelper to explain the parsing.

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
From [Issue: 1232](https://github.com/dotnet/coreclr/issues/1232)
> **@Clockwork-Muse wrote:**
>That code makes me nervous; maybe it should be broken up? Although if the current form was for performance reasons, perhaps comments are in order.

I've included addition comments to try an explain what the parsing code is doing. Along with some extra information as to what the error actually are.

### CODE REVIEW

**Comment by AdamSpeight2008** [comment]
Second commit: Extract out the limits (Index and Width) to public accessible static readonly properties.

File: src/mscorlib/src/System/Text/StringBuilder.cs [hunk 1]
```diff
@@ -1308,7 +1308,10 @@ void ISerializable.GetObjectData(SerializationInfo info, StreamingContext contex
         private static void FormatError() {
             throw new FormatException(Environment.GetResourceString("Format_InvalidString"));
         }
-        
+
+        public static readonly int Index_Limit = 1000000;
+        public static readonly int Width_Limit = 1000000;
```
Comment (weshaggard) [inline src/mscorlib/src/System/Text/StringBuilder.cs]: We do not add new public APIs without going through the [API review process](https://github.com/dotnet/corefx/blob/master/Documentation/project-docs/api-review-process.md)

Comment (AdamSpeight2008) [inline src/mscorlib/src/System/Text/StringBuilder.cs]: @weshaggard Hence two commits, first without and second with. They could be `private`

**Comment by adityamandaleeka** [comment]
@weshaggard @AlexGhiondea 

Ping... I was just triaging through some old PRs and saw that this is still open after almost a year. Is there interest in accepting the change with the added comments?

Personally, I think some of them don't add value:

```csharp
// Is the character a comma
if (ch == ',') {
```

I'll leave the decision to you guys. If this isn't something we want to take, please close the PR.

**Comment by AdamSpeight2008** [comment]
@adityamandaleeka 
This PR was mainly about documenting the actual exceptions and the undocumented value limits when using `.Format` methods. As well as explaining what parts of the algorithm or for.

**Comment by AdamSpeight2008** [comment]
Pinging @weshaggard @AlexGhiondea for signs of PR Life.

**Comment by weshaggard** [comment]
Sorry for the long delay this just slipped through my Radar. @AdamSpeight2008 I agree with @adityamandaleeka that some of the comments aren't very useful and can just be removed. I also don't want these constants to be public. I'm OK if you make them private.

**Comment by AdamSpeight2008** [comment]
@weshaggard  I've made a few changes to the comments and made the values constants.

**Comment by weshaggard** [comment]
Thanks for the contribution @AdamSpeight2008

**Pull request merged**